src/packageurl/__init__.py

```diff
@@ -459,12 +467,17 @@ def from_string(cls, purl: str) -> Self:
             url=remainder, scheme="", allow_fragments=True
         )
 
-        if scheme or authority:
```

Comment:
This would be a serious API change, why do you remove this?

Reply:
Because it was causing things to blow up when trying to de-serialize an SBOM which had PURLs with colons in the name. Given the discussion here: https://github.com/package-url/packageurl-python/issues/152 and specifically the comment here: https://github.com/package-url/packageurl-python/issues/152#issuecomment-2637692538 I think it is reasonable to enable the de-serialization of name and namespaces with `:` in them.